lib/testcase.js

```diff
@@ -185,7 +190,9 @@ TestCase.prototype._run = function () {
   if (this._payload) req.write(this._payload)
   req.end()
 
-  var promise = defer.promise
+  var promise = setupPromise.then(function () {
```

Comment:
I did not, and that is a bug. I missed the fact that the HTTP request fires off immediately even if the code extracts the promise from it later. -> Round 3.